packages/vow/src/when.js

```diff
@@ -28,8 +28,14 @@ export const makeWhen = (
     let result = await specimenP;
     let payload = getVowPayload(result);
     let priorRetryValue;
+    const seenPayloads = new WeakSet();
     while (payload) {
-      result = await basicE(payload.vowV0)
+      const { vowV0 } = payload;
```

Comment:
Leaving as a TODO for later, I don't want to impede the pragmatic goal of getting this PR merged.

TODO: encapsulate this extraction in `vow-utils.js` as a subfunction used by `toPassableCap`, but taking a payload instead of a possible Vow:

```js
/**
 * comments here...
 * @param {VowPayload}
 * @returns {PassableCap}
 */
export const vowPayloadToPassableCap = payload => payload.vowV0;
```

Doing this would require more thought to decide when we actually should depend on `vowV0` vs when we only want a passable cap that can serve as a store key.